buildbot/master.cfg

```diff
@@ -42,33 +32,43 @@ WEBAUTH_PASSWORD = open("webauth-password.txt").read().strip()
 
 OVHPROXY_URL = "https://ovhproxy.dolphin-emu.org/?url="
 
-class GitNoBranch(Git):
-    """Monkey patch for stupid patch, --branch and fetch behavior."""
+class DolphinGit(buildstep.BuildStep, buildstep.ShellMixin, buildstep.CommandMixin):
+    def __init__(self, **kwargs):
+        kwargs = self.setupShellMixin(kwargs)
+        buildstep.BuildStep.__init__(self,
+                                     name="git",
+                                     description="updating",
+                                     descriptionDone="update",
+                                     haltOnFailure=True,
+                                     **kwargs)
+
+    @defer.inlineCallbacks
+    def shell(self, **kwargs):
+        cmd = yield self.makeRemoteShellCommand(**kwargs)
+        yield self.runCommand(cmd)
+        if cmd.didFail():
+            raise buildstep.BuildStepFailed()
+
+    @defer.inlineCallbacks
+    def run(self):
+        git_exists = yield self.pathExists("build/.git")
+
+        if not git_exists:
+            yield self.shell(workdir=".", command=["git", "clone", "--progress", "https://github.com/dolphin-emu/dolphin.git", "build"])
+
+        if self.getProperty("pr_id"):
+            ref = "pull/%s/head" % self.getProperty("pr_id")
```

Comment:
The only problem with merge is that last time I checked, it was updated asynchronously after the PR is submitted / updated.
If it takes too long, then we will rebuild the old changes again.